packages/block-library/src/social-link/variations.js

```diff
@@ -203,108 +224,126 @@ const variations = [
 		attributes: { service: 'meetup' },
 		title: 'Meetup',
 		icon: MeetupIcon,
+		patterns: [ /^https:\/\/(www\.)?meetup\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'medium',
 		attributes: { service: 'medium' },
 		title: 'Medium',
 		icon: MediumIcon,
+		patterns: [ /^https:\/\/medium\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'patreon',
 		attributes: { service: 'patreon' },
 		title: 'Patreon',
 		icon: PatreonIcon,
+		patterns: [ /^https:\/\/(www\.)?patreon\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'pinterest',
 		attributes: { service: 'pinterest' },
 		title: 'Pinterest',
 		icon: PinterestIcon,
+		patterns: [ /^https:\/\/(www\.)?pinterest\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'pocket',
 		attributes: { service: 'pocket' },
 		title: 'Pocket',
 		icon: PocketIcon,
+		patterns: [ /^https:\/\/getpocket\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'reddit',
 		attributes: { service: 'reddit' },
 		title: 'Reddit',
 		icon: RedditIcon,
+		patterns: [ /^https:\/\/(www\.)?reddit\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'skype',
 		attributes: { service: 'skype' },
 		title: 'Skype',
 		icon: SkypeIcon,
+		patterns: [ /^https:\/\/join\.skype\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'snapchat',
 		attributes: { service: 'snapchat' },
 		title: 'Snapchat',
 		icon: SnapchatIcon,
+		patterns: [ /^https:\/\/(www\.)?snapchat\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'soundcloud',
 		attributes: { service: 'soundcloud' },
 		title: 'SoundCloud',
 		icon: SoundCloudIcon,
+		patterns: [ /^https:\/\/soundcloud\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'spotify',
 		attributes: { service: 'spotify' },
 		title: 'Spotify',
 		icon: SpotifyIcon,
+		patterns: [ /^https:\/\/open\.spotify\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'telegram',
 		attributes: { service: 'telegram' },
 		title: 'Telegram',
 		icon: TelegramIcon,
+		patterns: [ /^https:\/\/t\.me(\/.+)?$/i ],
 	},
 	{
 		name: 'threads',
 		attributes: { service: 'threads' },
 		title: 'Threads',
 		icon: ThreadsIcon,
+		patterns: [ /^https:\/\/(www\.)?threads\.net(\/.+)?$/i ],
 	},
 	{
 		name: 'tiktok',
 		attributes: { service: 'tiktok' },
 		title: 'TikTok',
 		icon: TiktokIcon,
+		patterns: [ /^https:\/\/(www\.)?tiktok\.com(\/.+)?$/i ],
 	},
 	{
 		name: 'tumblr',
 		attributes: { service: 'tumblr' },
 		title: 'Tumblr',
 		icon: TumblrIcon,
+		patterns: [ /^https:\/\/(.+)\.tumblr\.com\/?(.+)?/i ],
 	},
 	{
 		name: 'twitch',
 		attributes: { service: 'twitch' },
 		title: 'Twitch',
 		icon: TwitchIcon,
+		patterns: [ /^https:\/\/(www\.)?twitch\.tv(\/.+)?$/i ],
 	},
 	{
 		name: 'twitter',
 		attributes: { service: 'twitter' },
 		title: 'Twitter',
 		icon: TwitterIcon,
+		patterns: [ /^https:\/\/twitter\.com(\/.+)?$/i ],
```

Comment:
I think the `twitter` variation should be left out of this, as it is considered a legacy variation and should not be receiving new features. [It was actually planned to be removed in 6.5.](https://github.com/WordPress/gutenberg/pull/54092#issuecomment-1713593014)

Reply:
That's probably best discussed separately, as it doesn't relate to this PR.

Reply:
It does relate, which is why I commented on the code diff. I'm just saying this line should be deleted or moved into the `x` variation.

Reply:
The domain may still be active but all of the on-page branding and favicon has changed to X. This is an icon block so it should match the iconography of the link destination.

Reply:
Updated in [f083c83](https://github.com/WordPress/gutenberg/pull/59303/commits/f083c83864dc8ee649b7e3c6d3ae4843a71b9f2f).